### PR TITLE
Add GET /resume/{id} endpoint

### PR DESCRIPTION
**Pull request by shahshlok**
## 📝 Description

Adds a new `GET /resume/{id}` endpoint to retrieve a single resume item by its ID. This enables fetching individual resume items for detailed views or editing workflows.

**Key behaviors:**
- Returns the resume item with associated project name when found
- Returns 404 if the item doesn't exist
- Returns 404 if the associated project is soft-deleted (maintains data consistency)
- Returns the item with `project_name: null` for orphan resume items (no associated project)

**Closes:** #330 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

- [x] All 12 tests in `tests/api/test_resume_by_id.py` pass
- [x] Tested success case: `GET /resume/1` returns correct resume item
- [x] Tested 404 case: nonexistent ID returns `{"detail": "Resume item not found"}`
- [x] Tested soft-delete case: resume items with deleted projects return 404
- [x] Tested orphan case: resume items without projects return with `project_name: null`
- [x] Tested invalid ID: string IDs return 422 validation error

Run tests with:
```bash
uv run pytest tests/api/test_resume_by_id.py -v
```

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [ ] 🔗 Any dependent changes have been merged and published in downstream modules
- [ ] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

Not applicable - API endpoint only, no UI changes.

### CODE REVIEW

**Comment by shahshlok** [comment]
Lines 158-193 is what I have added, all other changes are a result of formatting and linting the code using ruff and thus they dont need to be reviewed

File: src/artifactminer/api/retrieval.py [hunk 1]
```diff
@@ -146,6 +155,42 @@ async def get_resume_items(
...
+async def get_resume_item_by_id(
+    resume_id: int,
+    db: Session = Depends(get_db),
+) -> ResumeItemResponse:
+    """Retrieve a single resume item by its ID.
+
+    Returns 404 if the item doesn't exist or its associated project is soft-deleted.
+    Orphan items (no associated project) are returned with project_name: null.
+    """
+    resume_item = db.query(ResumeItem).filter(ResumeItem.id == resume_id).first()
```
Comment (Whiteknight07) [inline src/artifactminer/api/retrieval.py]: use `.one_or_none()` instead of `.first()` - better reflects uniqueness expectation for id lookup and catches duplicate anomalies

File: src/artifactminer/api/retrieval.py [hunk 2]
```diff
@@ -146,6 +155,42 @@ async def get_resume_items(
...
+
+    if resume_item is None:
+        raise HTTPException(status_code=404, detail="Resume item not found")
+
+    # Check if associated project is soft-deleted
+    if resume_item.repo_stat_id is not None:
+        repo_stat = (
```
Comment (Whiteknight07) [inline src/artifactminer/api/retrieval.py]: n+1 query: this is a second db round trip. consider collapsing into single query with `outerjoin` on RepoStat + filter `deleted_at.is_(None)` like the list endpoint does

File: src/artifactminer/api/retrieval.py [hunk 3]
```diff
@@ -146,6 +155,42 @@ async def get_resume_items(
...
+            db.query(RepoStat).filter(RepoStat.id == resume_item.repo_stat_id).first()
+        )
+        if repo_stat is None or repo_stat.deleted_at is not None:
```
Comment (Whiteknight07) [inline src/artifactminer/api/retrieval.py]: good: proper soft-delete protection prevents data leakage for deleted projects

File: tests/api/test_resume_by_id.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+    """Returns 422 for invalid ID format (e.g., string instead of int)."""
+    response = client_with_data.get("/resume/not-a-number")
+
+    assert response.status_code == 422
+
+
+# === GET /resume/{id} Edge Cases ===
+
+
+def test_get_resume_by_id_soft_deleted_project(client_with_soft_deleted_project):
```
Comment (Whiteknight07) [inline tests/api/test_resume_by_id.py]: good: covers soft-deleted project case

**Review by Whiteknight07 — CHANGES_REQUESTED** [review]
overall solid implementation with good soft-delete handling and test coverage. main gaps: extra db query and unclear contract for invalid ids. requesting changes for the items below.

**Review by van-cpu — APPROVED** [review]
Overall really good work, covers edge cases and empty database cases. All tests pass for me as well.
